samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -467,25 +464,30 @@ object SamzaContainer extends Logging {
             } else {
               null
             }
+
             val keySerde = config.getStorageKeySerde(storeName) match {
               case Some(keySerde) => serdes.getOrElse(keySerde,
                 throw new SamzaException("StorageKeySerde: No class defined for serde: %s." format keySerde))
               case _ => null
             }
+
             val msgSerde = config.getStorageMsgSerde(storeName) match {
               case Some(msgSerde) => serdes.getOrElse(msgSerde,
                 throw new SamzaException("StorageMsgSerde: No class defined for serde: %s." format msgSerde))
               case _ => null
             }
-            val storeBaseDir = if(changeLogSystemStreamPartition != null) {
+
+            val storeDir = if (changeLogSystemStreamPartition != null) {
               TaskStorageManager.getStorePartitionDir(loggedStorageBaseDir, storeName, taskName)
-            }
-            else {
+            } else {
               TaskStorageManager.getStorePartitionDir(defaultStoreBaseDir, storeName, taskName)
             }
+
+            storeWatchPaths.add(storeDir.toPath)
```

Comment:
Exactly, now it adds only the stores currently used by the tasks in the container.